framework/src/governance/lib/datazone/datazone-gsr-msk-datasource.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Duration, Stack } from 'aws-cdk-lib';
+import { CfnProjectMembership } from 'aws-cdk-lib/aws-datazone';
+import { Rule, RuleTargetInput, Schedule } from 'aws-cdk-lib/aws-events';
+import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
+import {
+  Effect,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { Function, Runtime, Code } from 'aws-cdk-lib/aws-lambda';
+import { Construct } from 'constructs';
+import { DataZoneGsrMskDataSourceProps } from './datazone-gsr-msk-datasource-props';
+import { TrackedConstruct, TrackedConstructProps } from '../../../utils';
+
+/**
+ * A DataZone custom data source for MSK (Managed Streaming for Kafka) with integration for Glue Schema Registry.
+ *
+ * @example
+ * import { Schedule } from 'aws-cdk-lib/aws-events';
+ *
+ * new dsf.governance.DataZoneGsrMskDataSource(this, 'MskDatasource', {
+ *   domainId: 'aba_dc999t9ime9sss',
+ *   projectId: '999999b3m5cpz',
+ *   registryName: 'MyRegistry',
+ *   clusterName: 'MyCluster',
+ *   runSchedule: Schedule.cron({ minute: '0', hour: '12' }), // Trigger daily at noon
+ *   enableSchemaRegistryEvent: true, // Enable events for Glue Schema Registry changes
+ * });
+ */
+export class DataZoneGsrMskDataSource extends TrackedConstruct {
+
+  /**
+   * The IAM Role of the Lambda Function interacting with DataZone API
+   */
+  public readonly datasourceLambdaRole: Role;
+  /**
+   * The membership of the Lambda Role on the DataZone Project
+   */
+  public readonly dataZoneMembership: CfnProjectMembership;
+  /**
+   * The Event Bridge Rule for schema creation and update
+   */
+  public readonly createUpdateEventRule?: Rule;
+  /**
+   * The Event Bridge Rule for trigger the data source execution
+   */
+  public readonly scheduleRule?: Rule;
+  /**
+   * The Event Bridge Rule for schema deletion
+   */
+  public readonly deleteEventRule?: Rule;
+
+  /**
+   * Build an instance of the DataZoneGsrMskDataSource
+   * @param scope the Scope of the CDK Construct
+   * @param id the ID of the CDK Construct
+   * @param props The DataZoneGsrMskDataSourceProps properties
+   */
+  constructor(scope: Construct, id: string, props: DataZoneGsrMskDataSourceProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: DataZoneGsrMskDataSource.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    const stack = Stack.of(this);
+    const accountId = stack.account;
+    const region = stack.region;
+    const partition = stack.partition;
+
+    const clusterArn = `arn:${partition}:kafka:${region}:${accountId}:cluster/${props.clusterName}/*`;
+    const listClustersArn = `arn:${partition}:kafka:${region}:${accountId}:/api/v2/clusters`;
```

Comment:
is this the right resource ARN?
I do not see it in the AWS IAM [reference](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonmanagedstreamingforapachekafka.html) for msk